lib/webtask.js

```diff
@@ -218,9 +224,16 @@ exports.issue_token = function (options /* req, claims, kid, keyset, store_code_
             new_claims.ten = new_claims.ten.regex.toString();
     }
 
-    // Encrypt context
 
-    if (new_claims.ectx) {
+    if(new_claims.ectx) {
+        // Token ectx stores authorization values correctly
+
+        if(!(new_claims.ectx.WEBTASK_JWT_SECRET && new_claims.ectx.WEBTASK_JWT_AUD) &&
+            (new_claims.ectx.WEBTASK_JWT_SECRET || new_claims.ectx.WEBTASK_JWT_AUD))
```

Comment:
I think the logical check we are after here is a negated exclusive or: `!(!!new_claims.ectx.WEBTASK_JWT_SECRET ^ !!new_claims.ectx.WEBTASK_JWT_AUD)`